domain/src/main/kotlin/feature/BackupManager.kt

```diff
@@ -0,0 +1,58 @@
+package ltd.evilcorp.domain.feature
+
+import android.content.res.Resources
+import android.net.Uri
+import androidx.core.os.ConfigurationCompat
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import ltd.evilcorp.core.repository.MessageRepository
+import ltd.evilcorp.core.vo.Message
+import java.text.DateFormat
+import java.text.SimpleDateFormat
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.collections.List
+
+@Singleton
+class BackupManager @Inject constructor(
+    private val messageRepository: MessageRepository,
+) {
+
+    companion object {
+        private val mapper = jacksonObjectMapper()
+    }
+
+    private fun List<Message>.generateBackupMessages(locationSave: String): BackupMessages {
+        return BackupMessages(
+            version = 1, // TODO @Akito: Increment version programmatically on major changes.
+            timestamp = SimpleDateFormat(
+                """yyyy-MM-dd'T'HH-mm-ss""",
+                ConfigurationCompat
+                    .getLocales(Resources.getSystem().configuration).get(0),
+            ).format(Date()),
+            locationSaved = locationSave,
+            entries = this,
+        )
+    }
+
+    private fun List<Message>.generateBackupMessagesJString(locationSave: String): String =
+        mapper.writerWithDefaultPrettyPrinter().writeValueAsString(this.generateBackupMessages(locationSave))
+
+    private fun getMessages(publicKey: String): List<Message> =
```

Comment:
This helper doesn't add anything. Just call messageRepostory.getStatic where you need the messages.